hydrogym/core.py

```diff
@@ -355,19 +422,43 @@ def step(
     """Advance the state of the environment.  See gym.Env documentation
 
         Args:
-            action (Iterable[ArrayLike], optional): Control inputs. Defaults to None.
+            action (Iterable[ActType], optional): Control inputs. Defaults to None.
 
         Returns:
-            Tuple[ArrayLike, float, bool, dict]: obs, reward, done, info
+            Tuple[ObsType, float, bool, dict]: obs, reward, done, info
         """
-    self.solver.step(self.iter, control=action)
-    self.iter += 1
-    t = self.iter * self.solver.dt
+    action = action * self.flow.CONTROL_SCALING
+
+    if self.num_sim_substeps_per_actuation is not None and self.num_sim_substeps_per_actuation > 1:
```

Comment:
If I understand correctly the `num_sim_substeps_per_actuation` allows to `stretch the duration of the actions uniformly (actions given to the damped controller are piecewise constants). For this piecewise action signal, `num_sim_substeps_per_actuation` adjusts the length of the intervals on which the action is constant?